sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/batch/implementation/BatchAsyncContainerExecutor.java

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.batch.implementation;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosBridgeInternal;
+import com.azure.cosmos.ThrottlingRetryOptions;
+import com.azure.cosmos.batch.TransactionalBatchOperationResult;
+import com.azure.cosmos.batch.TransactionalBatchResponse;
+import com.azure.cosmos.implementation.AsyncDocumentClient;
+import com.azure.cosmos.implementation.HttpConstants.HttpHeaders;
+import com.azure.cosmos.implementation.RequestOptions;
+import com.azure.cosmos.implementation.ResourceThrottleRetryPolicy;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.directconnectivity.WFConstants.BackendHeaders;
+import com.azure.cosmos.implementation.routing.CollectionRoutingMap;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import io.netty.util.HashedWheelTimer;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Semaphore;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkState;
+import static com.azure.cosmos.implementation.routing.PartitionKeyInternalHelper.getEffectivePartitionKeyString;
+
+/**
+ * Bulk batch executor for operations in the same container.
+ * <p>
+ * It maintains one {@link BatchAsyncStreamer} for each Partition Key Range, which allows independent execution of
+ * requests. Semaphores are in place to rate limit the operations at the Streamer / Partition Key Range level, this
+ * means that we can send parallel and independent requests to different Partition Key Ranges, but for the same Range,
+ * requests will be limited. Two delegate implementations define how a particular request should be executed, and how
+ * operations should be retried. When the {@link BatchAsyncStreamer} dispatches a batch, the batch will create a request
+ * and call the executeAsync delegate, if conditions are met, it might call the retry delegate.
+ * <p>
+ * {@link BatchAsyncStreamer}
+ */
+public class BatchAsyncContainerExecutor implements AutoCloseable {
+
+    private static final int DEFAULT_MAX_DEGREE_OF_CONCURRENCY = 150;
```

Comment:
Why is this one different from C#? https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Batch/BatchAsyncContainerExecutor.cs#L40

Reply:
Ack. Will modify it. Thanks.